create/serviceconnection.go

```diff
@@ -0,0 +1,213 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/alecthomas/kong"
+	meta "github.com/ninech/apis/meta/v1alpha1"
+	networking "github.com/ninech/apis/networking/v1alpha1"
+	"github.com/ninech/nctl/api"
+)
+
+type serviceConnectionCmd struct {
+	resourceCmd
+	Source                   string                    `placeholder:"kubernetescluster/my-clustername" help:"The source of the connection in the form kind/name. Allowed source kinds are: ${allowed_sources}." required:""`
```

Comment:
What about adding a `Reference` type and implementing the `encoding.TextUnmarshaler` or `json.Unmarshaler`:

https://github.com/alecthomas/kong?tab=readme-ov-file#custom-decoders-mappers

This would make the code a bit cleaner IMO and easier to reuse.